src/components/tabController/TabPage.tsx

```diff
@@ -49,9 +51,14 @@ export default function TabPage({
   useAnimatedReaction(() => {
     return targetPage.value === index;
   },
-  isActive => {
+  (isActive, wasActive) => {
     if (isActive) {
       runOnJS(lazyLoad)();
+      runOnJS(setFocused)(true);
+    }
+
+    if (wasActive && wasActive !== isActive) {
```

Comment:
Why not `wasActive && !isActive ` or (to be sure we don't have a race condition):
```
if (isActive) {
  ...
} else if (wasActive) {
  runOnJS(setFocused)(false);
}
```

Reply:
I change it.. now it's better. 
Apparently we got to this part of code more than once, after adding `currentPage` to dependency array of `useAnimatedReaction` it fixed it.. so we can rely completely on `wasActive`

Reply:
@M-i-k-e-l 
I changed the implementation again, I think it's even simpler now
It has a single `useAnimatedReaction` so I think it's more bug prune 
Let me know if you find something 😬